sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1219,7 +1241,8 @@ class Analyzer(
 
       // Make sure a plan's expressions do not contain outer references
       def failOnOuterReference(p: LogicalPlan): Unit = {
-        if (p.expressions.exists(containsOuter)) {
+        p.expressions.foreach(checkMixedReferencesInsideAggregation)
+        if (!p.isInstanceOf[Filter] && p.expressions.exists(containsOuter)) {
```

Comment:
@viirya This is by choice. So we want to return the same error for TC.01.03 and TC.01.04. Currently we return a "..not supported outside of WHERE/HAVING" error for TC.01.04 which is misleading.